scala/private/rule_impls.bzl

```diff
@@ -85,28 +82,28 @@ def compile_scala(
     args.add("--StatsfileOutput", statsfile)
     args.add("--EnableDiagnosticsReport", enable_diagnostics_report)
     args.add("--DiagnosticsFile", diagnosticsfile)
-    args.add_joined("--Classpath", compiler_classpath_jars, join_with = ctx.configuration.host_path_separator)
-    args.add_joined("--ClasspathResourceSrcs", classpath_resources, join_with = ",")
-    args.add_joined("--Files", sources, join_with = ",")
-    args.add_joined("--Plugins", plugins, join_with = ",")
-    args.add_joined("--ResourceTargets", [p[0] for p in resource_paths], join_with = ",")
-    args.add_joined("--ResourceSources", [p[1] for p in resource_paths], join_with = ",")
-    args.add_joined("--ResourceJars", resource_jars, join_with = ",")
-    args.add_joined("--ScalacOpts", scalacopts, join_with = ":::")
-    args.add_joined("--SourceJars", all_srcjars, join_with = ",")
+    args.add_all("--Classpath", compiler_classpath_jars)
+    args.add_all("--ClasspathResourceSrcs", classpath_resources)
+    args.add_all("--Files", sources)
+    args.add_all("--Plugins", plugins)
+    args.add_all("--ResourceTargets", [p[0] for p in resource_paths])
```

Comment:
@simuons can you explain why do we need `--ResourceTargets`?

Reply:
`ResourceSources` are files to be included and `ResourceTargets` are paths under which they need to be included in final jar (with removed prefixes and workspace relative parts). This is handled https://github.com/bazelbuild/rules_scala/blob/master/scala/private/resources.bzl#L2 
And yes there is that weird contract that positions need to match in distinct lists.